### ISSUE TITLE
 TypeError: undefined is not an object (evaluating 'navigationState.routes.map')

### ISSUE THREAD

**Issue by bogdan124** [react-native-community/react-native-tab-view#495]
22:51:08: TypeError: undefined is not an object (evaluating 'navigationState.routes.map')

This error is located at:
    in TabViewAnimated (at Boiler.js:190)
    in TabViewExample (at SceneView.js:17)
    in SceneView (at CardStack.js:455)
    in RCTView (at View.js:71)
    in View (at CardStack.js:454)
    in RCTView (at View.js:71)
    in View (at CardStack.js:453)
    in RCTView (at View.js:71)
    in View (at createAnimatedComponent.js:147)
    in AnimatedComponent (at Card.js:12)
    in Card (at PointerEventsContainer.js:39)
    in Container (at CardStack.js:498)
    in RCTView (at View.js:71)
    in View (at CardStack.js:414)
    in RCTView (at View.js:71)
    in View (at CardStack.js:413)
    in CardStack (at CardStackTransitioner.js:67)
    in RCTView (at View.js:71)
    in View (at Transitioner.js:142)
    in Transitioner (at CardStackTransitioner.js:19)
    in CardStackTransitioner (at StackNavigator.js:41)
    in Unknown (at createNavigator.js:13)
    in Navigator (at createNavigationContainer.js:226)
    in NavigationContainer (at registerRootComponent.js:35)
    in RootErrorBoundary (at registerRootComponent.js:34)
    in ExpoRootComponent (at renderApplication.js:35)
    in RCTView (at View.js:71)
    in View (at AppContainer.js:102)
    in RCTView (at View.js:71)
    in View (at AppContainer.js:122)
    in AppContainer (at renderApplication.js:34)

I keep geting that error this is my code:

```
import MapView from 'react-native-maps';
import * as React from 'react';
import { AppRegistry,View, StyleSheet, Dimensions,Image,Text } from 'react-native';
import { TabViewAnimated, TabBar, SceneMap } from 'react-native-tab-view';
import { StackNavigator } from "react-navigation";
const initialLayout = {
  height: 150,
  width: Dimensions.get('window').width,
};



/*
constructor(props){
  super(props){
    this.state={
initialPosition:{
  latitude: 37.78825,
  longitude: -122.4324,
  latitudeDelta: 0.0922,
  longitudeDelta: 0.0421,
},
markerPosition:{
  latitude: 37.78825,
  longitude: -122.4324,
}

    }
  }
}
*/


const FirstRoute = () =>(



 <MapView
  style={styles.map}
initialRegion={this.state.initialPosition}>

<MapView.Marker
coordinate={this.state.markerPosition}>



<View style={styles.radius}>
<View style={styles.marker}>

</View>
</View>

</MapView.Marker>

</MapView>
)

//{{
//  latitude: 43.6526594,
//  longitude: 25.3634145,
  //latitudeDelta: 0.0922,
//  longitudeDelta: 0.0421,
//}}
//  latitude:  43.6526594,
//  longitude: 25.3634145,
//}}

const SecondRoute = () => <Text style={styles.pulse_data}  >65</Text>
const ThirdRoute = () => <View style={[ styles.container, { backgroundColor: '#673ab7' } ]} />;

const {width,height}=Dimensions.get('window')

const SCREEN_HEIGHT=height
const SCREEN_WIDTH=width

const ASPECT_RATIO=width/height

const LATTITUDE_DELTA = 0.092
const LONGITUDE_DELTA = LATTITUDE_DELTA * ASPECT_RATIO

export default class TabViewExample extends React.Component {
  constructor(props){

    super(props)

      this.state={

  initialPosition:{
    latitude:0,
    longitude: 0,
    latitudeDelta: 0,
    longitudeDelta: 0
  },
  markerPosition:{
    latitude: 0,
    longitude: 0
  }

      }
    }

  static navigationOptions = {
  //  headerStyle: {
  //   headerMode: 'none' ,
  //  },
  header: null,
  title: 'Login',

  };
  state1 = {
    index: 0,
    routes: [
      { key: 'first', title: 'Map' },
      { key: 'second', title: 'Pulse' },
      { key: 'third', title: 'Third' },
    ],
  };

  _handleIndexChange = index => this.setState({ index });

  _renderHeader = props => <TabBar {...props} />;

  _renderScene = SceneMap({
    first: FirstRoute,
    second: SecondRoute,
    third: ThirdRoute,
  });

  componentDidMount() {
    navigator.geolocation.getCurrentPosition(
       (position) => {
         console.log("wokeeey");
         console.log(position);
         this.setState({
           latitude: parseFloat(position.coords.latitude),
           longitude: parseFloat(position.coords.longitude),
           error: null,
         });
         var initialRegion = {

           latitude:position.coords.latitude,
           longitude:position.coords.longitude,
           latitudeDelta:LATTITUDE_DELTA,
           longitudeDelta:LONGITUDE_DELTA,
         }
         this.setState({initialPosition:initialRegion})
         this.setState({markerPosition:initialRegion})
       },
       (error) => this.setState({ error: error.message }),
       { enableHighAccuracy: false, timeout: 200000, maximumAge: 1000 },
     );


    this.watchId = navigator.geolocation.watchPosition(
        (Position) => {
          console.log("wokeeey");
          console.log(Position);
          this.setState({
            latitude: parseFloat(Position.coords.latitude),
            longitude: parseFloat(Position.coords.longitude),
            error: null,
          });
          var lastRegion = {

            latitude:Position.coords.latitude,
            longitude:Position.coords.longitude,
            latitudeDelta:LATTITUDE_DELTA,
            longitudeDelta:LONGITUDE_DELTA,
          }
          this.setState({initialPosition:lastRegion})
          this.setState({markerPosition:lastRegion})
        },
        (error) => this.setState({ error: error.message }),
        { enableHighAccuracy: false, timeout: 20000, maximumAge: 1000 },
      );



     }



   componentWillUnmount(){
     navigator.geolocation.clearWatch(this.watchId )
   }
  render() {
    return (

      <TabViewAnimated
        style={styles.container}
        navigationState={this.state}
        renderScene={this._renderScene}
        renderHeader={this._renderHeader}
        onIndexChange={this._handleIndexChange}
        initialLayout={initialLayout}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop:24,
  },
  map:{
    height:'100%',
    width:'100%',
  },
  pulse_data:{
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        marginTop:250,
        marginLeft:190,
        //height:'100%',
        //width:'100%',
        fontSize:40,
        //backgroundColor:"blue",

  },
});

AppRegistry.registerComponent("Boiler", () => Boiler);
```

**Comment by qbig** [comment]
``` {javascript}
<TabViewAnimated
        style={styles.container}
        navigationState={this.state}
        renderScene={this._renderScene}
        renderHeader={this._renderHeader}
        onIndexChange={this._handleIndexChange}
        initialLayout={initialLayout}
      />

```
`navigationState={this.state}` should be  `navigationState={this.state1}` instead ...

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
